static/app/components/arithmeticBuilder/token/function.tsx

```diff
@@ -154,36 +312,101 @@ function InternalInput({
         })
       );
     });
-  }, [attributesFilter, functionArguments, getFieldDefinition]);
+  }, [attributesFilter, builderFunctionArguments, getFieldDefinition]);
 
-  const items = useAttributeItems({
+  const attributeItems = useAttributeItems({
     allowedAttributes,
     filterValue,
   });
 
-  const shouldCloseOnInteractOutside = useCallback(
-    (el: Element) => !rowRef.current?.contains(el),
-    [rowRef]
-  );
+  const items = useMemo(() => {
+    // If this is a dropdown parameter, use the predefined options
+    if (parameterDefinition?.kind === 'dropdown' && parameterDefinition.options) {
+      return parameterDefinition.options
+        .filter(
+          option =>
+            !filterValue ||
+            option.value.toLowerCase().includes(filterValue.toLowerCase()) ||
+            option.label?.toLowerCase().includes(filterValue.toLowerCase())
+        )
+        .map(option => ({
+          key: option.value,
+          label: option.label ?? option.value,
+          value: option.value,
+          textValue: option.value,
+          hideCheck: true,
+        }));
+    }
+
+    // Otherwise, use the attribute-based items
+    return attributeItems;
+  }, [parameterDefinition, filterValue, attributeItems]);
+
+  const shouldCloseOnInteractOutside = useCallback((el: Element) => {
+    return !gridCellRef.current?.contains(el);
+  }, []);
 
   const onClick = useCallback(() => {
     updateSelectionIndex();
   }, [updateSelectionIndex]);
 
   const onInputBlur = useCallback(() => {
     resetInputValue();
+    setIsCurrentlyEditing(false);
   }, [resetInputValue]);
 
+  const onTextInputBlur = useCallback(() => {
+    if (hasNextArgument) {
+      focusTarget(
+        argumentsListState,
+        argumentsListState.collection.getKeyAfter(argumentItem.key)
+      );
+      if (inputValue) {
+        onArgumentsChange(argumentIndex, inputValue);
+      }
```

Comment:
Does this branch ever fire? I guess if you were to take the other comment about getting rid of the dropdown variant, it might be needed but this is a strange place to change focus to the next argument.

Reply:
I can't remember why I added this if block, I can't think of a good reason! I'm going to remove it.